src/v2/Utils/Hooks/useLazyLoadComponent.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect, useState } from "react"
+import { LazyLoadComponent } from "react-lazy-load-image-component"
+
+const Ready = ({ onMount }) => {
+  useEffect(onMount, [onMount])
+  return null
+}
+
+/**
+ * Simple hooks wrapper around LazyLoadComponent.
+ * Allows one to swap out components when mounted.
+ * TODO: Replace this with a normal intersection observer hook.
+ */
+export const useLazyLoadComponent = () => {
+  const [isMounted, setIsMounted] = useState(false)
+
+  const Sentinel = () => (
```

Comment:
`Sentinel` feels a bit obscure naming wise -- is there something slightly more verbose we could use?

Reply:
`Observer`?

Reply:
(Sentinel in the sense that it just watches and waits to be tripped.)

Reply:
Maybe `Waypoint`(which aligns with common terminology already used elsewhere)?

Reply:
Ah yeah that makes sense.